content/blog/2017/08/2017-08-07-intro-to-plugin-development.adoc

```diff
@@ -0,0 +1,68 @@
+* link:https://wiki.jenkins.io/display/JENKINS/Understanding+Jelly+Tags[Understanding Jelly Tags] from the Jenkins wiki
+* link:https://wiki.jenkins.io/display/JENKINS/Form+Validation[Form Validation] from the Jenkins wiki
+* link:https://wiki.jenkins.io/display/JENKINS/Jelly+form+controls[Jelly Form Controls] from the Jenkins wiki
+* link:http://reports.jenkins.io/reports/core-taglib/jelly-taglib-ref.html[Jelly Tag Library Reference] from jenkins.io
+* DataBoundConstructor in link:https://wiki.jenkins.io/display/JENKINS/Basic+guide+to+Jelly+usage+in+Jenkins[Basic Guide to Jelly usage]
+* DataBoundSetter in link:https://groups.google.com/d/msg/jenkinsci-dev/58-DEvuJZWI/5QrxBZRFJ6IJ[google groups]
+
+=== Testing a plugin
+
+* link:https://wiki.jenkins.io/display/JENKINS/Unit+Test[Unit test] from the Jenkins wiki
+* Jenkins test objects like link:http://javadoc.jenkins.io/archive/jenkins-1.642/org/jvnet/hudson/test/JenkinsRule.html[JenkinsRule] and the link:http://javadoc.jenkins.io/archive/jenkins-1.642/org/jvnet/hudson/test/WithoutJenkins.html[WithoutJenkins annotation]
+* DataBoundConstructor in link:https://wiki.jenkins.io/display/JENKINS/Basic+guide+to+Jelly+usage+in+Jenkins[Basic Guide to Jelly usage]
+* DataBoundSetter in link:https://groups.google.com/d/msg/jenkinsci-dev/58-DEvuJZWI/5QrxBZRFJ6IJ[google groups]
+* Java unit testing tools like link:https://code.google.com/archive/p/hamcrest/wikis/Tutorial.wiki[Hamcrest] and link:http://joel-costigliola.github.io/assertj/assertj-core-quick-start.html[AssertJ] (and link:http://javadoc.jenkins.io/archive/jenkins-1.642/org/jvnet/hudson/test/JenkinsMatchers.html[JenkinsMatchers])
+* Java unit testing rules like link:http://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html[TemporaryFolder], link:http://junit.org/junit4/javadoc/4.12/org/junit/rules/Timeout.html[Timeout], and link:http://junit.org/junit4/javadoc/4.12/org/junit/rules/DisableOnDebug.html[DisableOnDebug]
+* Java unit testing classes like link:http://junit.org/junit4/javadoc/4.12/org/junit/Assume.html[Assume] and link:http://junit.org/junit4/javadoc/4.12/org/junit/runners/Parameterized.html[Parameterized]
+* Java unit testing mock frameworks like link:http://site.mockito.org/[mockito] and link:http://powermock.github.io/[powermock]
+
+=== Javadoc
+
+* link:http://javadoc.jenkins.io/[Jenkins core javadoc]
+* link:http://javadoc.jenkins.io/plugin/[Jenkins plugins javadoc]
+
+=== Extension Points
+
+* List of link:https://jenkins.io/doc/developer/extensions/jenkins-core/[Jenkins core extension points] and link:https://jenkins.io/doc/developer/extensions/[all Jenkins extension points]
+
+=== Custom build steps
+
+* Adding a link:https://wiki.jenkins.io/display/JENKINS/Create+a+new+Plugin+with+a+custom+build+Step[custom build step]
+
+=== Actions
+
+* Jenkins link:https://www.cloudbees.com/blog/jenkins-internal-action-and-its-subtypes[Action and its subtypes]
```

Comment:
I think the end should include a call to action talking about your session and how the reader can learn a lot more if they grab one of the few remaining slots. Be sure to include the `JWFOSS` discount code like [recent posts](https://jenkins.io/blog/2017/07/27/standardizing-builds-with-shared-libraries/)